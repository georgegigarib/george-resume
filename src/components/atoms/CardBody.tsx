import { cn } from '@/utils/classMerge';
import React from 'react';

export const CardBody = ({ children, className }: { children: React.ReactNode; className?: string }) => {
  return (
    <div className={cn('h-96 w-96 [transform-style:preserve-3d] [&>*]:[transform-style:preserve-3d]', className)}>
      {children}
    </div>
  );
};
