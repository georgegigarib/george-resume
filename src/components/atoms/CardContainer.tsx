import { useRef, useState, createContext } from "react";
import { cn } from "../../utils/utils";


const MouseEnterContext = createContext<[boolean, React.Dispatch<React.SetStateAction<boolean>>] | undefined>(undefined);


export const CardContainer = ({ children, className, containerClassName }: { children?: React.ReactNode; className?: string; containerClassName?: string }) => {
    const containerRef = useRef<HTMLDivElement>(null);
    const [isMouseEntered, setIsMouseEntered] = useState(false);
  
    const handleMouseMove = (e: React.MouseEvent<HTMLDivElement>) => {
      if (!containerRef.current) return;
      const { left, top, width, height } = containerRef.current.getBoundingClientRect();
      const x = (e.clientX - left - width / 2) / 150;
      const y = (e.clientY - top - height / 2) / 100;
      containerRef.current.style.transform = `rotateY(${x}deg) rotateX(${y}deg)`;
    };
  
    const handleMouseEnter = () => {
      setIsMouseEntered(true);
      if (!containerRef.current) return;
    };
  
    const handleMouseLeave = () => {
      if (!containerRef.current) return;
      setIsMouseEntered(false);
      containerRef.current.style.transform = `rotateY(0deg) rotateX(0deg)`;
    };
    return (
      <MouseEnterContext.Provider value={[isMouseEntered, setIsMouseEntered]}>
        <div
          className={cn('flex w-full h-full', containerClassName)}
          style={{
            perspective: '1000px',
          }}
        >
          <div
            ref={containerRef}
            onMouseEnter={handleMouseEnter}
            onMouseMove={handleMouseMove}
            onMouseLeave={handleMouseLeave}
            className={cn('w-full relative transition-all duration-200 ease-linear', className)}
            style={{
              transformStyle: 'preserve-3d',
            }}
          >
            {children}
          </div>
        </div>
      </MouseEnterContext.Provider>
    );
  };