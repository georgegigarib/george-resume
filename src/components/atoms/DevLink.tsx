import React from 'react'
import { Sections } from '@/app/layout/navbarTypes'
import { useTranslation } from 'react-i18next'

const DevLink = ({
  scrollToSection,
  setAutomaticScroll,
  activeHash,
}: {
  scrollToSection: (id: string, called?: boolean) => void
  setAutomaticScroll: React.Dispatch<React.SetStateAction<boolean>>
  activeHash: string
}) => {
  const { t } = useTranslation()

  return (
    <a
      onClick={() => {
        scrollToSection(Sections.Dev, true)
        setAutomaticScroll(false)
      }}
      className={`block rounded md:p-0 ${
        activeHash === Sections.Dev
          ? 'text-white font-bold animated-transform translate-y-[15%] md:translate-y-[45%] scale-130'
          : activeHash === Sections.Music
            ? '-translate-y-[60%] md:-translate-y-[80%] 2xl:-translate-y-[50%] animated-transform'
            : 'text-gray-900 animated-transform rotate-[18deg] -translate-y-[20%] sm:rotate-12 md:translate-y-[0%] xl:rotate-3 opacity-40 font-bold 2xl:translate-y-[20%]'
      }`}
      aria-current={activeHash === Sections.Dev ? 'page' : undefined}
    >
      {t('navbar.dev')}
    </a>
  )
}

export default DevLink
