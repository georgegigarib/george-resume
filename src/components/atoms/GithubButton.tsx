import { SOCIAL_LINKS } from '@/constants/info'
import { useIsMobile } from '@/hooks/useIsMobile'
import GitHubIcon from '@mui/icons-material/GitHub'
import { Tooltip } from '@mui/material'

export default function GitHubButton() {
  const ismobile = useIsMobile()

  return (
    <Tooltip title="GitHub" arrow>
      <a
        href={SOCIAL_LINKS.github}
        target="_blank"
        className="p-1 md:p-2 flex justify-center items-center rounded-full bg-gray-100 dark:bg-gray-700 text-gray-700 dark:text-gray-300 hover:bg-gray-200 dark:hover:bg-gray-600 transition-colors"
      >
        <GitHubIcon fontSize={ismobile ? 'small' : 'medium'} />
      </a>
    </Tooltip>
  )
}
