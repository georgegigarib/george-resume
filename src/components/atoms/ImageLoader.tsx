import React, { useState } from 'react'
import CircularProgress from '@mui/material/CircularProgress'

interface ImageWithLoaderProps {
  imagePath: string
  imgClassName?: string
  style?: React.CSSProperties
  dataTestId?: string
  alt?: string
  onClick?: () => void
}

const ImageWithLoader: React.FC<ImageWithLoaderProps> = ({
  imagePath,
  imgClassName,
  style,
  dataTestId,
  alt,
  onClick,
}) => {
  const [isLoaded, setIsLoaded] = useState(false)

  return (
    <>
      {!isLoaded && (
        <div className="absolute inset-0 flex items-center justify-center">
          <CircularProgress />
        </div>
      )}
      <img
        src={imagePath}
        alt={alt}
        className={`${imgClassName} ${isLoaded ? 'opacity-100' : 'opacity-0'}`}
        style={style}
        data-testid={dataTestId}
        onLoad={() => setIsLoaded(true)}
        onClick={onClick}
      />
    </>
  )
}

export default ImageWithLoader
