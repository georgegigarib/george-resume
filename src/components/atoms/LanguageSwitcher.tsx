import { Menu, MenuItem, Button } from '@mui/material'
import { useState } from 'react'
import ChangeLanguageIcon from '@mui/icons-material/Language'
import ArrowDropDown from '@mui/icons-material/ArrowDropDown'
import { useLanguage } from '@/hooks/useLanguage'
import { languages, LanguageOption } from '@/locales/languages'
import i18n from '@/locales/i18n.config'

const LanguageSwitcher = () => {
  const { changeLanguage } = useLanguage()
  const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null)

  const languageOptions: LanguageOption[] = Object.values(languages)

  const currentLanguage = languages[i18n.language as keyof typeof languages]

  return (
    <>
      <div
        className="flex items-center justify-between w-full gap-10"
        data-testid="language-switcher"
      >
        <Button
          variant="contained"
          onClick={(event) => setAnchorEl(event.currentTarget)}
          style={{
            borderRadius: '20px',
            display: 'flex',
            alignItems: 'center'
          }}
          size="small"
          data-testid="language-switcher-button"
        >
          <ChangeLanguageIcon style={{ marginRight: '8px' }} />
          {currentLanguage?.name || 'English'}
          <ArrowDropDown
            style={{
              marginLeft: 'auto',
              transform: anchorEl ? 'rotate(180deg)' : 'rotate(0deg)',
              transition: 'transform 0.3s ease'
            }}
          />
        </Button>

        <Menu
          anchorEl={anchorEl}
          open={Boolean(anchorEl)}
          onClose={() => setAnchorEl(null)}
          PaperProps={{
            style: {
              boxShadow: '0px 4px 10px rgba(0, 0, 0, 0.4)',
              border: '1px solid rgba(0, 0, 0, 0.1)',
              borderRadius: '5px'
            }
          }}
          data-testid="language-menu"
        >
          {languageOptions.map((lang) => (
            <MenuItem
              key={lang.code}
              onClick={() => {
                changeLanguage(lang.code)
                setAnchorEl(null)
              }}
              data-testid={`language-option-${lang.code}`}
            >
              {lang.name}
            </MenuItem>
          ))}
        </Menu>
      </div>
    </>
  )
}

export default LanguageSwitcher
