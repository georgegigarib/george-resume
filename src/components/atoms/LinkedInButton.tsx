import { SOCIAL_LINKS } from '@/constants/info'
import { useIsMobile } from '@/hooks/useIsMobile'
import LinkedInIcon from '@mui/icons-material/LinkedIn'
import { Tooltip } from '@mui/material'

export default function LinkedInButton() {
  const ismobile = useIsMobile()

  return (
    <Tooltip title="LinkedIn" arrow>
      <a
        href={SOCIAL_LINKS.linkedin}
        target="_blank"
        rel="noopener noreferrer"
        className="p-1 md:p-2 flex justify-center items-center rounded-full bg-blue-100 dark:bg-blue-900 text-blue-600 dark:text-blue-300 hover:bg-blue-200 dark:hover:bg-blue-800 transition-colors"
      >
        <LinkedInIcon fontSize={ismobile ? 'small' : 'medium'} />
      </a>
    </Tooltip>
  )
}
