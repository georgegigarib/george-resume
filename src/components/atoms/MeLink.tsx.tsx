import { Sections } from '@/app/layout/navbarTypes'
import React from 'react'
import { useTranslation } from 'react-i18next'

const MeLink = ({
  scrollToSection,
  setAutomaticScroll,
  activeHash
}: {
  scrollToSection: (id: string, called?: boolean) => void
  setAutomaticScroll: React.Dispatch<React.SetStateAction<boolean>>
  activeHash: string
}) => {
  const { t } = useTranslation()

  return (
    <a
      onClick={() => {
        scrollToSection(Sections.Me, true)
        setAutomaticScroll(false)
      }}
      className={`block pt-8 rounded md:p-0 ${
        activeHash === Sections.Me
          ? 'text-white font-bold animated-transform -translate-y-[50%] md:translate-y-[50%] scale-130'
          : 'text-gray-900 -translate-y-[75%] sm:-translate-y-[70%] animated-transform md:dark:hover:bg-transparent opacity-40 font-bold'
      } ${
        activeHash === Sections.Music
          ? 'rotate-[18deg] sm:rotate-6 md:-translate-y-[10%] 2xl:translate-y-[15%] lg:rotate-3'
          : activeHash === Sections.Dev
            ? '-rotate-[18deg] sm:-rotate-6  md:translate-y-[0%]  2xl:translate-y-[15%] lg:-rotate-3'
            : ''
      }`}
    >
      {t('navbar.me')}
    </a>
  )
}

export default MeLink
