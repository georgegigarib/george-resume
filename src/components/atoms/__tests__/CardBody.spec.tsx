import { render, screen } from '@testing-library/react'
import { describe, expect, it } from 'vitest'
import { CardBody } from '@/components/atoms/CardBody'

describe('CardBody component', () => {
  it('matches the snapshot', () => {
    const { asFragment } = render(
      <CardBody className="custom-class">
        <p>Test content</p>
      </CardBody>
    )
    expect(asFragment()).toMatchSnapshot()
  })

  it('renders children correctly', () => {
    render(
      <CardBody className="custom-class">
        <p>Test content</p>
      </CardBody>
    )
    const content = screen.getByText('Test content')
    expect(content).toBeTruthy()
  })

  it('applies additional className', () => {
    const { container } = render(
      <CardBody className="custom-class">
        <p>Test content</p>
      </CardBody>
    )
    const div = container.firstChild as HTMLElement
    expect(div).toHaveClass('custom-class')
  })

  it('has the default styles applied', () => {
    const { container } = render(
      <CardBody>
        <p>Test content</p>
      </CardBody>
    )
    const div = container.firstChild as HTMLElement
    expect(div).toHaveClass('h-96', 'w-96', '[transform-style:preserve-3d]', '[&>*]:[transform-style:preserve-3d]')
  })
})
