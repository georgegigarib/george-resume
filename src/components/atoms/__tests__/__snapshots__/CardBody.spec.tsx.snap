// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CardBody component > matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="h-96 w-96 [transform-style:preserve-3d] [&>*]:[transform-style:preserve-3d] custom-class"
  >
    <p>
      Test content
    </p>
  </div>
</DocumentFragment>
`;
